### ISSUE TITLE
[frontend]  S3Error: 503 in ml-pipeline-ui-artifact

### ISSUE THREAD

**Issue by lightning-like** [kubeflow/pipelines#6839]
### How did you deploy Kubeflow Pipelines (KFP)?
We are attempting to deploy the kubeflow pipeline over a local cent-os cluster using kustomize

We use multi-user isolation

k8s v1.19.6

On a custom Kubernetes deployment.
KFP version:
1.7.0
KFP SDK version:
kfp 1.8.6
kfp-pipeline-spec 0.1.11
kfp-server-api 1.7.0

### Steps to reproduce

For create run we use  `V2 compatible` mode and  default  `package_root`

Try open/load/view any artefact in runs gui 

```
Failed to get object in bucket mlpipeline at path v2/artifacts/pipeline/Georgy test/025b7274-585b-4aac-95c3-9ca2d609d47e/irs/output_dataset: S3Error: 503
```

Remove the ?namespace in the link then we can download the artefact.

log from  ml-pipeline-ui-artifact when we try get artefact with ?namespace
```
GET /pipeline/artifacts/get?source=minio&peek=256&bucket=mlpipeline&key=v2%2Fartifacts%2Fpipeline%2FGeorgy+test%2F025b7274-585b-4aac-95c3-9ca2d609d47e%2Firs%2Fmarkdown_artifact
Getting storage artifact at: minio: mlpipeline/v2/artifacts/pipeline/Georgy test/025b7274-585b-4aac-95c3-9ca2d609d47e/irs/markdown_artifact
S3Error: 503
    at getError (/server/node_modules/minio/dist/main/transformers.js:138:15)
    at /server/node_modules/minio/dist/main/transformers.js:158:14
    at DestroyableTransform._flush (/server/node_modules/minio/dist/main/transformers.js:80:10)
    at DestroyableTransform.prefinish (/server/node_modules/readable-stream/lib/_stream_transform.js:138:10)
    at DestroyableTransform.emit (events.js:223:5)
    at prefinish (/server/node_modules/readable-stream/lib/_stream_writable.js:619:14)
    at finishMaybe (/server/node_modules/readable-stream/lib/_stream_writable.js:627:5)
    at endWritable (/server/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at DestroyableTransform.Writable.end (/server/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at IncomingMessage.onend (_stream_readable.js:693:10) {
  code: 'UnknownError',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null
}
```

log from  ml-pipeline-ui when we try get artefact without ?namespace
```
GET /pipeline/artifacts/minio/mlpipeline/v2/artifacts/pipeline/Georgy%20test/025b7274-585b-4aac-95c3-9ca2d609d47e/irs/output_dataset
Getting storage artifact at: minio: mlpipeline/v2/artifacts/pipeline/Georgy test/025b7274-585b-4aac-95c3-9ca2d609d47e/irs/output_dataset
```

### Materials and Reference

We'll be grateful if you can get any clue what we can look more.

We tried to:
- env according #6750 and  #4649

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by lightning-like** [comment]
problem solved.
We fixed destination rule in Istio

**Comment by holadepo** [comment]
@lightning-like could you please describe what you changed in Istio. I have the same issue.

**Comment by pwzhong** [comment]
@holadepo have you fixed the problem? I am facing the same issue.
@lightning-like  Could you explain a bit more what istio changes you made?

**Comment by holadepo** [comment]
> @holadepo have you fixed the problem? I am facing the same issue. @lightning-like Could you explain a bit more what istio changes you made?

I have not fixed it yet. Please let me know if you have a fix

**Comment by yingding** [comment]
I have the same problem on an on-prem KF 1.4 with multi-tenancy. I manage to copy the secret `ml-pipeline-ui-artifact` from `kubeflow` namespace into my profile namespace, sothat the pipeline in my namespace can run. But the UI can't load `ml-pipeline-ui-artifact` from default minio. I got both `503` and `504` errors. Is there any workarround available, without upgrade to KFP 1.8.1?

**Comment by Bartket** [comment]
@yingding 

Add these lines to the ml-pipeline-ui-artifact deployment yaml file that you have within the namespace that you want to have this capability:

```
env:
  - name: MINIO_ACCESS_KEY
    valueFrom:
      secretKeyRef:
        name: mlpipeline-minio-artifact
        key: accesskey
  - name: MINIO_SECRET_KEY
    valueFrom:
      secretKeyRef:
        name: mlpipeline-minio-artifact
        key: secretkey
```

The workaround is to keep doing that with each new user/namespace added (I guess it can be automated via few kubectl commands).
![obraz](https://user-images.githubusercontent.com/22456734/173168009-3b9ba3c4-2a5e-46d9-9cb7-2c22acbc2ba1.png)

**Comment by yingding** [comment]
@Bartket Thanks for you hint
> ... ml-pipeline-ui-artifact deployment yaml file that you have within the namespace ...

I just want to double check, `ml-pipeline-ui-artifact` deployment is a KF 1.5 new feature, which I don't have in both my namespace and also not in `kubeflow` namespace using KF 1.4 right?

The artifacts works on our KF 1.5 installation, but it doesn't work on the other KF 1.4, I was hoping to patch KF 1.4 without upgrading as workaround first.

**Comment by RakeshRaj97** [comment]
@lightning-like I am facing a very similar issue. Using an external on-prem minio server. Changed secrets in user profile namespaces. When I try to view the logs -- marked in the screenshot -- I get the response `Failed to get object in bucket mlpipeline at path artifacts/conditional-execution-pipeline-with-exit-handler-lpcnt/2022/06/17/conditional-execution-pipeline-with-exit-handler-lpcnt-2953709790/main.log: S3Error: 400` (Screenshot attached)

<img width="1154" alt="Screen Shot 2022-06-17 at 5 01 53 pm" src="https://user-images.githubusercontent.com/47710229/174244192-70a82492-afa1-4c31-be0b-eca284a87718.png">


<img width="1154" alt="Screen Shot 2022-06-17 at 5 05 25 pm" src="https://user-images.githubusercontent.com/47710229/174244305-0df3c5f9-bf16-4c97-b48d-2b3cab1061fc.png">


I can see the pipelines logs pushed to my external minio but cant be retrieved from UI. I need to manually sign in to my minio to view the logs everytime

<img width="943" alt="Screen Shot 2022-06-17 at 5 07 31 pm" src="https://user-images.githubusercontent.com/47710229/174244885-054a52a9-b9f7-41f9-8603-aa581b85c487.png">

Followed this tutorial https://blog.min.io/how-to-kubeflow-minio/

Logs from ml-pipeline-ui-artifact in kubeflow-user-example-com namespace

```
GET /pipeline/artifacts/minio/mlpipeline/artifacts/conditional-execution-pipeline-with-exit-handler-lpcnt/2022/06/17/conditional-execution-pipeline-with-exit-handler-lpcnt-2953709790/main.log
Getting storage artifact at: minio: mlpipeline/artifacts/conditional-execution-pipeline-with-exit-handler-lpcnt/2022/06/17/conditional-execution-pipeline-with-exit-handler-lpcnt-2953709790/main.log
S3Error: 400
    at getError (/server/node_modules/minio/dist/main/transformers.js:138:15)
    at /server/node_modules/minio/dist/main/transformers.js:158:14
    at DestroyableTransform._flush (/server/node_modules/minio/dist/main/transformers.js:80:10)
    at DestroyableTransform.prefinish (/server/node_modules/readable-stream/lib/_stream_transform.js:138:10)
    at DestroyableTransform.emit (events.js:400:28)
    at prefinish (/server/node_modules/readable-stream/lib/_stream_writable.js:619:14)
    at finishMaybe (/server/node_modules/readable-stream/lib/_stream_writable.js:627:5)
    at endWritable (/server/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at DestroyableTransform.Writable.end (/server/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at IncomingMessage.onend (internal/streams/readable.js:670:10) {
  code: 'UnknownError',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null
}
```

Can i get a clue on this?

**Comment by Bartket** [comment]
@yingding We are using Kubeflow 1.4 (Kubeflow on AWS version).

Fyi: You should be good by applying this fix in sync.py and redeploying via Kustomize. I am yet to test it.

[gh pr checkout 5864](https://github.com/kubeflow/pipelines/pull/5864/files/764840ccd7ae85dd4f4c497d84602c40519c51e9#diff-997ca857ae89736a948d7bc38a69264639b3ddebfaf0eb172d13d35f6f535d62)

**Comment by dvaldivia** [comment]
@RakeshRaj97 hey I wrote that tutorial hahaha glad someone is using it, I just hit the same problem you are facing while doing a deployment on OpenShift, this doesn't happen on other vanilla environments, did you found a solution to this problem?

**Comment by dvaldivia** [comment]
error was caused because the MinIO external artifact storage was behind TLS so specifying port `443` is not enough `MINIO_SSL="true"` as an environment variable to the pipelines ui deployment is also needed.

**Comment by RakeshRaj97** [comment]
@dvaldivia Thanks for that great tutorial. It is very helpful and saved me a ton of effort by centrally managing all data in one external minio. I highly appreciate and support your effort on making that tutorial and please continue to update it on a regular basis if possible.

MINIO_SSL='true' in `ml-pipeline-ui` deployment still not solving the issue. My current image for `ml-pipeline-ui` is `gcr.io/ml-pipeline/frontend:1.8.2`

I also see this error 
```
Failed to get object in bucket mlpipeline at path artifacts/create-pipeline-4sgrp/2022/10/18/create-pipeline-4sgrp-2895230044/main.log: Error: getaddrinfo ENOTFOUND internal-minio.example.com
```

**Comment by ksheerasagar17** [comment]
Im seeing the same issue on Kubeflow installation on Azure Kubernetes Service (AKS). Like the author said if we remove the ?namespace in the link then we can download the artefact. I tried all the above solutions but issue still continues.

I tried these 2 too but NOT working.

https://github.com/kubeflow/pipelines/issues/6839#issuecomment-1158605309
https://github.com/kubeflow/pipelines/issues/6839#issuecomment-1284719116

**Comment by luckyops** [comment]
It's work for me.

```yaml
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  labels:
    application-crd-id: kubeflow-pipelines
  name: ml-pipeline-minio
  namespace: kubeflow
spec:
  host: minio-service.kubeflow.svc.cluster.local
  trafficPolicy:
    tls:
      mode: DISABLE
```

**Comment by tugasakhirai21** [comment]
I installed Kubeflow on my on-premise Kubernetes cluster, and most things seem to be working fine. However, I'm encountering an issue when trying to access artifacts from the Run dashboard. Whenever I click on any step and try to access the corresponding artifact link on the right side, I receive an "RBAC: access denied" message.

I can log in to Minio directly and access these artifacts manually. Additionally, I noticed that when I remove the namespace component from the URL, I can download the artifact files successfully.

I've tried some solutions mentioned in this forum, but none of them have worked for me so far.

Any insights or suggestions on how to resolve this issue?
